crates/saya/core/src/prover/program_input.rs

```diff
@@ -56,8 +56,11 @@ pub fn extract_messages(
         .flat_map(|t| t.execute_call_info.iter())
         .filter(|c| c.entry_point_type == EntryPointType::L1Handler)
         .map(|c| {
-            let message_hash =
-                compute_l1_message_hash(*c.caller_address, *c.contract_address, &c.calldata[..]);
+            let message_hash = compute_l2_to_l1_message_hash(
+                c.caller_address.into(),
+                c.contract_address.into(),
+                &c.calldata,
+            );
```

Comment:
@glihm should we use l2 -> l1 or l1 -> l2 here?

Reply:
In this specific case, it's `l1 -> l2`. 👍
In this context, `starknet` is the "l1" (settlement layer), and a--chain is the "l2".

Reply:
computing the l1 -> l2 message hash here is not doable bcs we can't get the `nonce` from the `TxExecInfo`. so adding some changes in #1986 #1987